### PR TITLE
Separate out kwargs in processor

### LINKED ISSUE #30106: Unused or unrecognized kwargs: padding, max_length, truncation.
### System Info

- `transformers` version: 4.39.2
- Platform: Linux-3.10.0-1160.99.1.el7.x86_64-x86_64-with-glibc2.31
- Python version: 3.10.13
- Huggingface_hub version: 0.22.2
- Safetensors version: 0.4.2
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Fine-tuning the CLIP model. Previously, I passed both text and images to the processor and specified some parameters for the text: padding, max_length, truncation. Everything was working fine before upgrading to version 4.39. However, after the upgrade, I encountered a warning that affects the training process. How should I resolve this? 
image

### Expected behavior

Do I need to separate the text and images for processing and use processor.tokenizer and processor.image_processor separately?

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Separates out the kwargs that are passed to the processor into tokenizer and image processor kwargs. 

This solves two issues:
* Removes warnings from validation checks that now happen in the image processor (nice proof they're working!)
* Prevents passing kwargs which have a name clash between the two processing classes

Fixes #30106

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/models/clip/processing_clip.py
Comment (molbap) [inline src/transformers/models/clip/processing_clip.py:1]: Much tidier, cool!
